src/modules/BootSupportCheck.rb

```diff
@@ -256,8 +267,10 @@ def GRUB
     def GRUB2
       ret = GRUB()
       # ensure that s390 have ext* partition for booting (bnc#873951)
-      ret &&= check_zipl_part if Arch.s390
-      ret &&= check_gpt_reserved_partition if Arch.x86_64
+      ret = check_zipl_part && ret if Arch.s390
```

Comment:
Precedence can be very tricky here. I would use a more explicit format so it's understandable at a first sight for average Joe.